Source/Core/Core/WatchDog.cpp

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "Core/WatchDog.h"
+
+#include "Common/Assert.h"
+#include "Common/ChunkFile.h"
+#include "Common/Logging/Log.h"
+#include "Common/SPSCQueue.h"
+
+#include "VideoCommon/VideoConfig.h"
+
+#include <algorithm>
+#include <cmath>
+
+WatchDog::Profiler g_watchdog;
+WatchDog::Profiler g_gpu_watchdog;
+
+void WatchDog::Profiler::Reset(const Marker& first_marker)
```

Comment:
Everything else early outs with the profiler check.  Should this as well for consistency?